src/service/cid-impl.js

```diff
@@ -121,32 +121,56 @@ export class Cid {
     return consent.then(() => {
       return viewerForDoc(this.ampdoc).whenFirstVisible();
     }).then(() => {
-      return getExternalCid(this, getCidStruct,
-          opt_persistenceConsent || consent);
+      const cidPromise = this.getExternalCid_(
+          getCidStruct, opt_persistenceConsent || consent);
+      // Getting the CID might involve an HTTP request. We timeout after 10s.
+      return timerFor(this.ampdoc.win)
```

Comment:
I was trying to compare the behaviors with before.
However  I found we would always create a new cid during timeout before [code](https://github.com/ampproject/amphtml/blob/master/src/service/cid-impl.js#L257),  but with the new change, if timeout, we don't return a cid. 
Is this correct?